src/Runner.Worker/ActionManager.cs

```diff
@@ -184,6 +201,8 @@ private async Task<PrepareActionsState> PrepareActionsRecursiveAsync(IExecutionC
                     {
                         continue;
                     }
+                    // Get the download info
+                    var downloadInfos = await GetDownloadInfoAsync(executionContext, repositoryActions);
```

Comment:
why did we move this down into the `foreach` loop? this would mean we will call the actions service multiple times to get the same response.

Reply:
That makes sense. I'd originally moved this when I was considering a different approach to how we handled the actions service call. Reverted!